### PR TITLE
feat(markdown-docx): add softbreak transformer - #397

### LINKED ISSUE #397: CiceroMark<->OOXML transformers
# Feature Request 🛍️

Incorporating the CiceroMark->OOXML transformer and improving the currently implemented OOXML->CiceroMark transformer.

## Use Case

It will allow the conversion of Docx files into CiceroMark JSON and vice versa. In addition, using this `Docx` files can be converted into different other formats like `PDF` or `HTML` as well.

## Possible Solution

The transformer for OOXML->CiceroMark is already implemented. It needs to be updated with some entities to allow full transformation.

The transformer for CiceroMark->OOXML is created in `cicero-word-add-in` branch. It needs to be transferred/transported here with some changes to allow the transformation.

## Detailed Description

Currently, the CiceroMark->OOXML transformer supports the following conversions:
| CiceroMark Entity  | OOXML Entity |
| --------  | -------- |
| Text      | `<w:t>` |
| Paragraph     | `<w:p>Content</w:p>` |
| Linebreak      | `<w:p/>` |
| Softbreak      | `<w:r><w:sym></w:r>` |1
| Emph     | `<w:i>` |
| Variable      | `<w:sdt>` |
| List Block/ List     | `<w:numPr><w:num w:val={ordered/unordered}/></w:numPr>` |
| List Item(Text)     | `<w:t/>` |
| List Item(Variable)     | `<w:sdt/>` |

Conversions which are left:

- `Formula / Ergo expressions`
- `Clause`
- `Conditional`
- `Optional`
- `Strong`
- `BlockQuote`
- `CodeBlock`
- `Code`
- `ThematicBreak`
- `Link`
- `Image`

In the left conversions, we need to decide which ones need major importance/priority and which can be given a lower priority. Furthermore, we also need to think about whether all these will be present in the contract (IMO, Code and CodeBlock generally won't occur in the contract)

### PR DESCRIPTION
**Pull request by K-Kumar-01**
Signed-off-by: k-kumar-01 <kushalkumargupta4@gmail.com>

Partial fix #397 

### Changes

- Softbreak Rule
- Check for softbreak in transformer and call rule
- Test for above(acceptance-of-delivery.json)

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

File: packages/markdown-docx/src/CiceroMarkToOOXML/rules.js [hunk 1]
```diff
@@ -109,4 +109,16 @@ const VARIABLE_RULE = (title, tag, value, type) => {
...
+ * @returns {string} OOXML for softbreak
+ */
+const SOFTBREAK_RULE = () => {
+    return `
+      <w:r>
+        <w:sym w:font="Calibri" w:char="2009" />
+      </w:r>
+    `;
+};
+module.exports = { TEXT_RULE, EMPHASIS_RULE, HEADING_RULE, VARIABLE_RULE, SOFTBREAK_RULE };
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXML/rules.js:124]: Insert a line break before this statement.

File: packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js [hunk 2]
```diff
@@ -24,13 +24,9 @@ const { checkRoundTripEquality } = require('./helper');
...
-
     for (const file of fileNames) {
-        // acceptance-of-delivery requires advance transformer. Skip for now.
-        if (file !== 'acceptance-of-delivery.json') {
-            it(`should parse ${file.replace('.json', '')}.`, async () => {
-                await checkRoundTripEquality(`${directoryName}/${file}`);
-            });
-        }
+        it(`should parse ${file.replace('.json', '')}.`, async () => {
+            await checkRoundTripEquality(`${directoryName}/${file}`);
```
Comment (algomaster99) [inline packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:29]: Rename `acceptance-of-delivery` to have softbreak in its name. The test cases will be clearer as we don't know what nodes the former contains.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
See comments.

File: packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js (hunk 2 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:29]: @algomaster99 
`accpetance-of-delivery` is also used in the `OoxmlTransformer`. So I didn't change the name. Should I create a new file for the `softbreak`?

Comment (algomaster99) [reply packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:29]: Creating a new file would be great. Thanks!

Comment (K-Kumar-01) [reply packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:29]: @algomaster99 Done

**Review by algomaster99 — APPROVED** [review]

**Pull request merged**